lib/internal/validators.js

```diff
@@ -13,6 +14,43 @@ function isUint32(value) {
   return value === (value >>> 0);
 }
 
+const octalReg = /^[0-7]+$/;
+const modeDesc = 'must be a 32-bit unsigned integer or an octal string';
+// Validator for mode_t (the S_* constants). Valid numbers or octal strings
+// will be masked with 0o777 to be consistent with the behavior in POSIX APIs.
+function validateAndMaskMode(value, name, def) {
+  if (isUint32(value)) {
+    return value & 0o777;
+  }
+
+  if (typeof value === 'number') {
+    if (!Number.isInteger(value)) {
+      throw new ERR_OUT_OF_RANGE(name, 'an integer', value);
+    } else {
+      // 2 ** 32 === 4294967296
+      throw new ERR_OUT_OF_RANGE(name, '>= 0 && < 4294967296', value);
+    }
+  }
+
+  if (typeof value === 'string') {
+    if (!octalReg.test(value)) {
```

Comment:
Why do we need this? If the number is not a parseable octal value, `NaN` will be returned, which we check for anyway.

Reply:
`parseInt` will not return `NaN` for strings like `'666abc'`.
But I agree there seems to be redundancy. Could there be cases where the string passes the regex and the value cannot be parsed? I don't think so.

Reply:
@mscdex @targos Yes there are existing test cases (`'123x'`) in our code base, it is currently only enforced in `process.umask` though. It does not seem quite right to remove it IMO. Passing a number followed by invalid characters looks like a mistake that the user may want to know.

Reply:
By the way, https://github.com/tc39/proposal-number-fromstring (currently in Stage 1) is designed for this situation

Reply:
@joyeecheung what I meant is that the `Number.isNaN(parsed)` check can be removed. Unless I'm missing an edge case, it can never be true after the regex check.